### DISCUSSION TITLE
How hard would it be to introduce support for sparsity?

### DISCUSSION THREAD

**Discussion by afg1** [ggml-org/llama.cpp#1759] [Ideas]
Given the recent publication of the [SpQR technique](https://arxiv.org/abs/2306.03078), and the further compression it will allow, is it worthwhile implementing sparsity here?

I get the impression that this is a seriously non-negligible amount of work, so maybe it doesn't fit. That being said, sparsity should make inference faster, and possible on smaller machines still, so maybe it is of interest?

**Comment by Green-Sky** [comment]
closing in favor of https://github.com/ggerganov/llama.cpp/discussions/2061
